OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -60,6 +61,14 @@ public void AddPalette(string name, ImmutablePalette p, bool allowModifiers)
 			int index = palettes.Count;
 			indices.Add(name, index);
 			palettes.Add(name, p);
+
+			if (palettes.Count > buffer.GetLength(0))
+			{
+				var oldBuffer = buffer;
+				buffer = new uint[Exts.NextPowerOf2(palettes.Count), Palette.Size];
```

Comment:
But I may be missing something, so if it does take more work than the couple of lines that i'm imagining then it would be better left to its own PR.